src/Microsoft.Azure.WebJobs.Host/Bindings/Path/BindingTemplate.cs

```diff
@@ -99,28 +129,26 @@ public string Bind(IReadOnlyDictionary<string, string> parameters)
                 parameters = caseInsensitive;
             }
 
+            StringBuilder builder = new StringBuilder();
             foreach (BindingTemplateToken token in Tokens)
             {
                 if (token.IsParameter)
                 {
-                    string value;
-                    BindingParameterResolver resolver = null;
-                    if (parameters != null && parameters.TryGetValue(token.Value, out value))
-                    {
-                        // parameter is resolved from binding data
-                        builder.Append(value);
-                    }
-                    else if (BindingParameterResolver.TryGetResolver(token.Value, out resolver))
+                    var resolverContext = new ParameterResolverContext
                     {
-                        // parameter maps to one of the built-in system binding
-                        // parameters (e.g. rand-guid, datetime, etc.)
-                        value = resolver.Resolve(token.Value);
-                        builder.Append(value);
-                    }
-                    else
+                        ParameterIndex = token.Index,
+                        ParameterName = token.Value,
+                        Template = _pattern,
+                        BindingData = parameters,
+                        Properties = bindingContext?.Properties
```

Comment:
Currently this Properties bag is the only reason BindingData needs to flow through. I couldn't add BindingData in its entirety on ParameterBindingContext because the core assembly doesn't have access to the host assembly. I think that is fine though.